src/jit/block.h

```diff
@@ -1171,6 +1161,16 @@ struct BasicBlock : private LIR::Range
 
     void MakeLIR(GenTree* firstNode, GenTree* lastNode);
     bool IsLIR();
+
+    void SetDominatedByExceptionalEntryFlag()
+    {
+        bbFlags |= BBF_DOMINATED_BY_EXCEPTIONAL_ENTRY;
+    }
+
+    bool IsDominatedByExceptionalEntryFlag()
+    {
+        return (bbFlags & BBF_DOMINATED_BY_EXCEPTIONAL_ENTRY) != 0;
+    }
```

Comment:
We add new EH only during fgFindBasicBlocks or inlining. It happens before computing dominators.